docs/README.md

```diff
@@ -166,6 +185,213 @@ sudo ip link del wg0
 
 Since the wireguard agents are userspace in both Windows and Darwin, the tunnel interface is removed when the agent process exits.
 
+### Deploying on Kubernetes managed Node
+
+Instructions mentioned in [Deploying on Node](#deploying-on-node) can be used here to deploy Apex agent on kubernetes managed nodes. However deploying the agent across all the nodes in a sizable Kubernetes cluster can be a challenging task, manually and even with scripting. Following section provides instructions to deploy Kubernetes style manifest to automate the deployment process.
+
+#### Setup the configuration
+
+Agent deployment in kubernetes requires few initial configuration details to successfully deploy the agent and onboard the node. This configuration is provided through [kustomization.yaml](../deploy/apex-client/overlays/dev/kustomization.yaml).
+
+Fetch the CA cert from the Kubernetes/Openshift Cluster that is running Apex Controller, and set the return cert string to `cert` literal of `secretGenerator` in `kustomization.yaml`.
+
+```sh
+kubectl get secret -n apex apex-ca-key-pair -o json | jq -r '.data."ca.crt"'
+```
+
+If you are not running your Apex service at the default DNS addresses set in the `kustomization.yaml` file, please update the `url` and `auth_url` accordingly. Update the username and password as well if your Apex Controller is configured with a different username and password.
+
+Note: Current username and password are default configuration, which is very likely to change in near future.
+
+Please update the `<APEX_CONTROLLER_IP>` with the IP address where the Apex controller is reachable.
```

Comment:
With Kind deployment it's the machine Ip right ? or are you thinking of any other scenario ?

Reply:
I will try deploying the agent on the Kind cluster (not sure how it will work in Container-in-Container environment), but I am wondering if deploying the agent within the apex cluster helps dev in anyways (specifically in Kind environment), because all the kind nodes are containers on the same node, and currently we are using kernel space wireguard. So i think to run any useful dev setup, dev needs atleast a single node Cluster (or microshift) for onboarding at the minimum.